lib/SILGen/SILGenDestructor.cpp

```diff
@@ -219,6 +220,135 @@ void SILGenFunction::destroyClassMember(SILLocation cleanupLoc,
   }
 }
 
+llvm::SmallSetVector<VarDecl*, 4> findRecursiveLinks(DeclContext* DC, ClassDecl *cd) {
+  auto SelfTy = DC->mapTypeIntoContext(cd->getDeclaredType());
```

Comment:
getDeclaredType() is deprecated:

```
  /// getDeclaredType - Retrieve the type declared by this entity, without
  /// any generic parameters bound if this is a generic type.
  ///
  /// Since UnboundGenericType is on its way out, so is this method. Try to
  /// avoid introducing new callers if possible. Instead of passing around
  /// an UnboundGenericType, considering passing around the Decl itself
  /// instead.
  Type getDeclaredType() const;
```

Also it never contains type parameters so mapTypeIntoContext() is a no-op. What you want instead is `SelfTy = cd->getDeclaredInterfaceType()`